airflow/jobs/base_job.py

```diff
@@ -110,6 +111,7 @@ def __init__(self, executor=None, heartrate=None, *args, **kwargs):
             self.heartrate = heartrate
         self.unixname = getuser()
         self.max_tis_per_query: int = conf.getint("scheduler", "max_tis_per_query")
+        get_listener_manager().hook.on_starting()
```

Comment:
```suggestion
        get_listener_manager().hook.on_starting(self)
```

(Because this is also called for LocalTaskJob, so being able to tell them apart is useful)

Reply:
Fixed.